### ISSUE TITLE
ServletContainerInitializer is passed invalid @HandlesTypes classes

### ISSUE THREAD

**Issue by mbakhoff** [OpenLiberty/open-liberty#16598]
**Describe the bug**

Servlet 3.0 javax.servlet.ServletContainerInitializer can be used to find classes that implement a specific interface. 

```
@HandlesTypes({MyInterface.class})
public class MyServletContainerInitializer implements ServletContainerInitializer {
  @Override
  public void onStartup(Set<Class<?>> scanResult, ServletContext ctx) throws ServletException {
    // scanResult should contain all implementations of MyInterface
  }
}
```

Open Liberty not only includes the implementations but also the interface itself. The spec says it should only include the implementations. 

I discovered this when my vaadin19 project failed to start up. 

**Steps to Reproduce**
Clone https://github.com/mbakhoff/openliberty-sci-handledtypes-poc, run `mvn liberty:dev` and open http://localhost:9080/sample/sci 

It will show `[class io.openliberty.sample.MyImpl, interface io.openliberty.sample.MyInterface]`

**Diagnostic information:**
 - OpenLiberty Version: 21.0.0.3
 - Java Version: AdoptOpenJDK 11
 - server.xml configuration: included in the sample repo linked above

**Additional context**
Same issue was present in WAS 8.5.5 https://www.ibm.com/support/pages/apar/PI26448

**Comment by pmd1nh** [comment]
Hi,

Thanks for open the issue and provide the test case.  It is highly appreciated.

This case is slightly different than what PI26448 fixed in WAS.  The key difference is that annotation has used in the previous reported APAR .  To confirm, I changed the provided test to use annotation, for example:

```
public @interface MyInterface 

and 

@MyInterface
public class MyImpl {
}
```

The response shows: [class io.openliberty.sample.MyImpl]


We are fixing this scenario in Liberty.  In the meantime, you can use the annotation as a workaround until an official fix is available.
 
Thanks!

**Comment by wtlucy** [comment]
@mbakhoff as an update, we're fixing this with https://github.com/OpenLiberty/open-liberty/pull/16759 - we decided to use the new _correct_ behavior with servlet 5.0+, and then to add a new webcontainer custom property `excludeAllHandledTypesClasses` to request the new behavior for servlet 4.0 and older.

**Comment by mbakhoff** [comment]
That's great news! Thank you!

**Comment by wtlucy** [comment]
OK, we're done with the code fix - if you'd like to test out the new behavior (by enabling `excludeAllHandledTypesClasses`) you can pull down a nightly build, otherwise it will be available in the upcoming 21.0.0.6 release.
